odoo/fields.py

```diff
@@ -2967,14 +2982,14 @@ def convert_to_display_name(self, value, record):
         return value.display_name if value else False
 
 
-class _Relational(Field):
+class _Relational(Field[_T]):
```

Comment:
```suggestion
class _Relational(Field[_T], Generic[_T]):
```

The class is still generic, so you need to add the `Generic` type (in the last position).
Same for other abstract types.

You also need to constrain T to a subtype of BaseModel.